### PR TITLE
feat: track parquet cache metrics

### LINKED ISSUE #25427: Add more parquet cache metrics
It would be useful to track some metrics on the parquet cache (see parent issue), e.g.,

* cache size - memory footprint (MB)
* cache size - number of files
* cache hits
* cache misses
* cache misses while loading

### PR DESCRIPTION
**Pull request by hiltontj**
Closes #25427

Adds metrics to track the following in the in-memory parquet cache:

* cache size in bytes (also included a fix in the calculation of that)
* cache size in n files
* cache hits
* cache misses
* cache misses while the oracle is fetching a file

A test was added to check this functionality

### CODE REVIEW

**Review by pauldix — APPROVED** [review]

**Comment by hiltontj** [comment]
I was reading through this and noticed some of the logic was a bit off in cache size calculations, so I corrected that in https://github.com/influxdata/influxdb/pull/25632/commits/147045a43700d8962dc8a57b89cdf3723003f993. Basically, I wasn't correctly accounting for the fact that a `Fetching` entry still has a size, which the code now does properly.

I also found some of the code and naming could be cleaned up so I did that as well.

**Comment by hiltontj** [comment]
Going to merge this to move things along, if there are any concerns in the commit I added after your approval @pauldix then  I can address in follow-on PRs.

**Pull request merged**
